provider.go

```diff
@@ -315,26 +342,27 @@ func (g *InstanceGroup) Increase(
 		})
 
 		if err != nil {
-			errs = append(errs, fmt.Errorf(
-				"failed to create %s: %w", serverName, err))
-			continue
-		}
-
-		server, err = g.client.Servers.WaitFor(
-			ctx, server.UUID, cloudscale.ServerIsRunning)
-
-		if err != nil {
-			errs = append(errs, fmt.Errorf(
-				"failed to wait for %s: %w", serverName, err))
+			errors_c <- fmt.Errorf("failed to create %s: %w", serverName, err)
 			continue
 		}
 
-		g.log.Info("created server", "name", serverName, "uuid", server.UUID)
-		servers = append(servers, server)
+		// Send the uuid to the "wait" goroutine
+		servers <- server.UUID
+	}
+	
+	// We wait for all previously started goroutines to complete before we return
+	wg.Wait()
+	// and we explicitly close the channel, so the waiting dispatcher
+	// (that for loop that recieves from the channel) can exit cleanly
+	close(servers)
```

Comment:
I think this should be closed as soon as the loop above is finished (probably necessary when you add the additional `wg.Add` call, but I think it's also good to close as soon as we do not expect new items to be added to a channel.

Reply:
done